expression/util.go

```diff
@@ -357,3 +357,80 @@ func extractFiltersFromDNF(ctx sessionctx.Context, dnfFunc *ScalarFunction) ([]E
 	}
 	return extractedExpr, ComposeDNFCondition(ctx, newDNFItems...)
 }
+
+// GetFuncArgLen gets the length of the row.
+func GetFuncArgLen(e Expression) int {
+	if f, ok := e.(*ScalarFunction); ok && f.FuncName.L == ast.RowFunc {
+		return len(f.GetArgs())
+	}
+	return 1
+}
+
+// CheckArgsOneColumn checks the arg has one column.
+func CheckArgsOneColumn(args ...Expression) error {
+	for _, arg := range args {
+		if GetFuncArgLen(arg) != 1 {
+			return ErrOperandColumns.GenByArgs(1)
+		}
+	}
+	return nil
+}
+
+// GetFuncArg gets the argument of the function at idx.
+func GetFuncArg(e Expression, idx int) Expression {
+	if f, ok := e.(*ScalarFunction); ok {
+		return f.GetArgs()[idx]
+	}
+	return nil
+}
+
+// PopFuncFirstArg pops the first element and return the rest of row.
+// e.g. After this function (1, 2, 3) becomes (2, 3).
+func PopFuncFirstArg(ctx sessionctx.Context, e Expression) (ret Expression, err error) {
```

Comment:
This func is only used when the function is RowFunc as the comment shows,
I think s/PopFuncFirstArg/ PopRowFuncFirstArg will be better.